### PR TITLE
Add login options to frontend: OSM or temp auth

### PR DESCRIPTION
**Pull request by NSUWAL123**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation
- [ ] 🧑‍💻 Refactor
- [ ] ✅ Test
- [ ] 🤖 Build or CI
- [ ] ❓ Other (please specify)

## Related Issue
- Related to #1438

## Describe this PR
This PR adds a functionality where on sign-in button click a popup is displayed where user will able to choose between OSM login or Temporary login.

## Screenshots
image

## Alternative Approaches Considered

Did you attempt any other approaches that are not documented in code?

## Review Guide

Notes for the reviewer. How to test this change?

## Checklist before requesting a review

- 📖 Read the FMTM Contributing Guide: <https://github.com/hotosm/fmtm/blob/main/CONTRIBUTING.md>
- 📖 Read the HOT Code of Conduct: <https://docs.hotosm.org/code-of-conduct>
- 👷‍♀️ Create small PRs. In most cases, this will be possible.
- ✅ Provide tests for your changes.
- 📝 Use descriptive commit messages.
- 📗 Update any related documentation and include any relevant screenshots.

### CODE REVIEW

File: src/frontend/src/assets/images/osm-logo.png
Comment (spwoodcock) [inline src/frontend/src/assets/images/osm-logo.png:1]: osm-logo

The logo is a big large for what we need, at 64kb or so.
Could we use the one attached, 64px x 64px and only 9kb?

**Review by spwoodcock — CHANGES_REQUESTED** [review]
Very clean, looks great!

**Comment by spwoodcock** [comment]
I assume the user is displayed as `svcfmtm` (or whatever service account was set up) after temporary login.
Could we possibly override this to always display as `Temporary`?

I know it's not super simple, as it probably also requires another state param to determine that the user logged in via temp auth, but it's a nicer user experience!

The temp login cookie (containing the token) lasts for 7 days.
The code in https://github.com/hotosm/fmtm/pull/1462 will detect if the cookie is no longer valid or present, and log the user out on the frontend automatically (clearing the login state).

**Comment by NSUWAL123** [comment]
@spwoodcock, currently when logged in with temporary account `local admin` is shown in the username. You mean to say that if a user is logging in from the temporary account, in the frontend we shall display username as `Temporary` instead of `local admin`?

**Comment by spwoodcock** [comment]
local admin is actually an override only used during development!

Comment out `DEBUG=true` in your .env file and run `docker compose up -d` to see what I mean. 

The related PR that I linked will change that behaviour. Maybe we should merge that first, then rebase this for testing before merge

**Comment by spwoodcock** [comment]
I just merged #1462 👍

**Review by spwoodcock — APPROVED** [review]
Looks good 👍 

It's easier for me to merge this on continue to work on it.

We might need a little tweak to make the `/auth/introspect` call on first page load play nicely with the temp auth/login.

**Pull request merged**